config/v1/types_infrastructure.go

```diff
@@ -87,6 +92,22 @@ const (
 	VSpherePlatformType PlatformType = "VSphere"
 )
 
+// PlatformStatus holds the current status specific to the underlying infrastructure provider
+// of the current cluster. Since these are used at status-level for the underlying cluster, it
+// is supposed that only one of the status structs is set.
+type PlatformStatus struct {
+	// AWS contains settings specific to the Amazon Web Services infrastructure provider.
+	// +optional
+	AWS *AWSPlatformStatus `json:"aws,omitempty"`
+}
+
+// AWSPlatformStatus holds the current status of the Amazon Web Services infrastructure provider.
+type AWSPlatformStatus struct {
+	// region holds the default AWS region for new AWS resources created by the cluster.
+	// +optional
+	Region string `json:"region,omitempty"`
```

Comment:
feels like this should be required if `AWS` is set - would we ever want to have a case where `Region` is not set?

Reply:
> would we ever want to have a case where `Region` is not set?

See the `The field is optional, because you only need to set it if you want the cluster to provision resources for you. For user-provided infrastructure flows...` paragraph in my f6189fedfaa commit message and PR topic.  Does that not hold water for you?

Reply:
I get that the AWS field would be optional. However if `aws` is set, I feel that allowing `region` to be empty/nil would provide zero benefit.

Reply:
> However if `aws` is set, I feel that allowing `region` to be empty/nil would provide zero benefit.

Just the "you don't have to bother setting this, the cluster can be ok without it" convenience benefit, so I'm still in favor of `+optional` here.  But the required -> optional transition is non-breaking while an optional -> required transition would be breaking, so I've pushed 30a62156 -> a1f3bdc3 dropping `+optional` and `omitempty`.  We can revisit later if we get push-back from UPI folks annoyed by the requirement ;).